lib/orchestrators/join.js

```diff
@@ -17,10 +17,11 @@ const join = (dispatch) => {
 
       // TODO better checking if task/join/junction/fork
       if (task.info) {
-        console.log('Joining to task: ' + task.info.name)
+        if (task.info.props) {
```

Comment:
we can consider using lodash.get for things like this (`_.get(task, 'info.props'`) since then we wont have runtime errors like `cannot get property props of undefined` (if `task.info` does not exist)

at the same time, all properties on the task state object should be declared in the initial state (i.e. what the default value is for `state` in the reducer `(state, action)`